src/app/services/date/date.service.ts

```diff
@@ -85,12 +102,36 @@ export class DateService {
 		const days = Array.from({ length: diff }, (_, i) => i + 1);
 
 		const dates = [start, ...days.map((additionalDays) => addDays(start, additionalDays))];
-
 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 		if (!isEqual(dates.at(-1)!, end)) {
+			console.error({ dates, end });
 			throw new Error('Unexpected getDates end mismatch');
 		}
 
 		return dates;
 	}
+
+	/**
+	 *
+	 * @param dateRangeDropdown value from DATE_RANGE_DROPDOWN_DATA
+	 * @returns A converted value from DATE_RANGE_DROPDOWN_DATA to a DateRange that
+	 * reference the [start, end] dates
+	 *
+	 * Used to receive errors from this.getDates() when not working with midnight dates
+	 */
+	getDateRangeByDropdown(dateRangeDropdown: DateRangeDropdown): DateRange {
```

Comment:
Is that bug where there is one extra day still here?

When picking 2 weeks, it went from wednesday from 2 weeks ago to wednesday,
this is actually 15 days:

Oct 19 (Wednesday), Oct 20, Oct 21, Oct 22, Oct 23, Oct 24, Oct 25, Oct 26, Oct 27, Oct 28, Oct 29, Oct 30, Oct 31, Nov 1, Nov 2 (Wednesday)

This is actually 15 days. Normally 2 weeks is 14 days

Reply:
bump

Reply:
the issue was resolved in [b869c6e](https://github.com/bitovi/opensource-analytics-dashboard/pull/15/commits/b869c6efa2f30cadc3e337e46be9ab1ea14ed76f) by adding +1 day after substracting 14 days from the start of the current day  by the following change:
`onst pastDate = addDays(subWeeks(startOfDay(new Date()), dateRangeDropdown.rangeValue), 1);`